OData/src/System.Web.OData/OData/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -391,7 +391,7 @@ private Expression CreateTotalCountExpression(Expression source, ExpandedNavigat
                 Expression propertyName = CreatePropertyNameExpression(elementType, propertyToExpand, source);
                 Expression propertyValue = CreatePropertyValueExpressionWithFilter(elementType, propertyToExpand, source,
                     expandItem.FilterOption);
-                Expression nullCheck = Expression.Equal(propertyValue, Expression.Constant(null));
+                Expression nullCheck = GetNullCheckExpression(propertyToExpand, propertyValue, projection);
```

Comment:
https://github.com/OData/WebApi/blob/f4573605bd88dabe882c4dc45e4a4ed0e55b32e0/OData/src/System.Web.OData/OData/Query/HandleNullPropagationOptionHelper.cs#L10

Can we check the LINQ provider and change the behavior?